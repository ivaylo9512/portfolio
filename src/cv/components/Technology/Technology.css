.technology{
    position: relative;
    font-size: 1vw;
    position: relative;
    display: inline-block;
    margin-right: 6.5%;
    text-align: center;
    width: 19.5%;
}
.technology .circle {
    filter: url(#shape-shadow);
    width: 100%;
    display: block;
}
.technology svg:last-child {
    position: absolute;
    width: 59%;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
}
.technology svg:nth-of-type(2){
    fill: black;
}
.angular{
    fill: black;
}
.angular path{
    fill: white;
}
.aws :nth-child(n + 3) {
    fill: white;
}
.typescript path{
    fill: white;
}
.technology svg:nth-of-type(2).react{
    fill: white;
}
.react g :first-child{
    fill: black;
}