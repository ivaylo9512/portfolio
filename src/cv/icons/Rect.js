const Rect = ({ height, width }) => {
    return(
        <svg className="rect" xmlns="http://www.w3.org/2000/svg" viewBox={`0 0 ${width} ${height}`} xmlnsXlink="http://www.w3.org/1999/xlink" >
            <rect width="100%" height="100%" rx="10" ry="10"/>
        </svg>
    )
}

export default Rect