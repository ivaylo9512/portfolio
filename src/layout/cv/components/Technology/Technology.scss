.technology{
    position: relative;
    font-size: 1vw;
    position: relative;
    display: inline-block;
    text-align: center;
    width: 15%;
    
    svg.circle {
        width: 100%;
        display: block;
    }

    .circle circle {
        filter: url(#shape-shadow);
    }

    svg:last-child {
        position: absolute;
        width: 59%;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
    }

    svg:nth-of-type(2){
        fill: black;
    }

    .angular{
        fill: black;
    }

    .angular path{
        fill: white;
    }

    .aws :nth-child(n + 3) {
        fill: white;
    }

    .typescript path{
        fill: white;
    }

    .technology svg:nth-of-type(2).react{
        fill: white;
    }

    .react g {
        fill: white;
    }

    @media only screen and (min-width: 700px) and (orientation:landscape){
        width: 21%;
        opacity: 1;
    }
}
