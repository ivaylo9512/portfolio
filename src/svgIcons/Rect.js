const Rect = ({ height, width, radius = 10 }) => {
    return(
        <svg className='rect' xmlns='http://www.w3.org/2000/svg' viewBox={`0 0 ${width} ${height}`} xmlnsXlink='http://www.w3.org/1999/xlink' >
            <rect width='100%' height='100%' rx={radius} ry={radius}/>
        </svg>
    )
}

export default Rect