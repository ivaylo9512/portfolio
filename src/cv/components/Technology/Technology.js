import Circle from 'cv/components/Circle/Circle';
import './Technology.css';

const Technology = ({ icon }) => {
    return(
        <div className='technology'>
            <Circle />
            {icon}
        </div>
    )
}

export default Technology