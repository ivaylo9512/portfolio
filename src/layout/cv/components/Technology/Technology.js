import Circle from '@svgIcons/Circle';
import './Technology.scss';

const Technology = ({ icon }) => {
    return(
        <div className='technology'>
            <Circle viewBoxUnit='129' />
            {icon}
        </div>
    )
}

export default Technology