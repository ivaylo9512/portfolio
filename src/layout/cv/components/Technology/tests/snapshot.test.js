import Technology from '../Technology';
import { shallow } from 'enzyme';
import Rect from '@svgIcons/Rect';

describe('Technology snapshot tests', () => {
    const createWrapper = () => shallow(
        <Technology icon={<Rect />}/>
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();
     
        expect(wrapper).toMatchSnapshot();
    })
});