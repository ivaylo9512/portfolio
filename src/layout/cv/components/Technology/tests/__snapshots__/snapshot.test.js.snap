// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology snapshot tests should match snapshot 1`] = `
<div
  className="technology"
>
  <Circle
    viewBoxUnit="129"
  />
  <Rect />
</div>
`;
